File: test_proxy.js
const axios = require('axios');
const { HttpsProxyAgent } = require('https-proxy-agent');

const proxy = 'http://brd-customer-hl_50cfa19c-zone-unodunoproxy:voo5cpt5qjsn@brd.superproxy.io:33335';
const agent = new HttpsProxyAgent(proxy);

async function test() {
  try {
    console.log("Testando proxy na Bright Data...");
    // Primeiro testamos o IP (que deve retornar um IP europeu/portugues)
    const res1 = await axios.get('https://lumtest.com/myip.json', { httpsAgent: agent });
    console.log("SUCESSO: Seu IP via Proxy é:", res1.data.ip);
    
    // Depois testamos o YouTube
    console.log("Testando acesso ao YouTube...");
    const res2 = await axios.get('https://www.youtube.com', { httpsAgent: agent });
    console.log("SUCESSO: YouTube acessado! Status:", res2.status);
    
  } catch (error) {
    if (error.response && error.response.status === 407) {
      console.error("ERRO 407: Autenticação falhou ou IP bloqueado (ip_forbidden).");
    } else {
      console.error("ERRO:", error.message);
    }
  }
}

test();

File: test_proxy.py
import requests

proxy_url = "http://brd-customer-hl_50cfa19c-zone-unodunoproxy:voo5cpt5qjsn@brd.superproxy.io:33335"
proxies = {
    "http": proxy_url,
    "https": proxy_url
}

print("Testando IP da Bright Data...")
try:
    res = requests.get("https://lumtest.com/myip.json", proxies=proxies, timeout=10)
    print("SUCESSO! IP do Proxy:", res.json().get('ip'))
except Exception as e:
    print("ERRO no IP:", e)

print("\nTestando acesso ao YouTube...")
try:
    res = requests.get("https://www.youtube.com", proxies=proxies, timeout=10)
    print("SUCESSO! Status YouTube:", res.status_code)
except Exception as e:
    print("ERRO no YouTube:", e)
